main.go

```diff
@@ -1,11 +1,36 @@
 package main
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	tf5server "github.com/hashicorp/terraform-plugin-go/tfprotov5/server"
+	tfmux "github.com/hashicorp/terraform-plugin-mux"
 	"github.com/hashicorp/terraform-provider-tfe/tfe"
 )
 
+const (
+	tfeProviderName = "registry.terraform.io/hashicorp/tfe"
+)
+
 func main() {
-	plugin.Serve(&plugin.ServeOpts{
-		ProviderFunc: tfe.Provider})
+	ctx := context.Background()
+	mux, err := tfmux.NewSchemaServerFactory(
+		ctx, tfe.Provider().GRPCProvider, tfe.PluginProviderServer,
+	)
+	if err != nil {
+		log.Println(fmt.Errorf("Could not setup a NewSchemaServerFactory using the providers: %v", err))
```

Comment:
1) I think you're going to want to prefix this log line with `[ERROR]`, so Terraform registers it as an error log.
2) I think you could use `log.Printf` here.